ssl/quic/quic_txp.c

```diff
@@ -2243,7 +2244,7 @@ static int txp_generate_stream_frames(OSSL_QUIC_TX_PACKETISER *txp,
 
         /* Choose between explicit or implicit length representations. */
         use_explicit_len = !((can_fill_payload || !could_have_following_chunk)
-                             && !needs_padding_if_implicit);
+                             && !pkt->force_pad);
```

Comment:
I considered doing it this way myself - but it wasn't clear to me whether it would work in all cases. The `pkt` structure only tells you about the current encryption level. Can it ever be that a PATH_RESPONSE frame has been emitted for a different encryption level in the datagram? In which case `pkt->force_pad` wouldn't be set here but we still need to use explicit len?

Reply:
Different ELs are represented by different packets, so there is no issue putting an implicit-length STREAM frame in different ELs, because they are implicitly ended by the end of the packet, not the end of the datagram.